### PR TITLE
migrate gatsby 1.0 to 2.0

### PR DESCRIPTION
**Pull request by Fabious**
Migration to Gatsby 2.0
-----------------------

Updated the breaking changes,
Bumped others deps too,
Refactor some code,
And fixed some wrong code (especially previous and next page on the docs)

If someone can test this feature at least locally i would be grateful, as i had to do a lot of changes ! :)

**Need to be fixed**
- weird behavior with the images when reloading a page (only in dev ?) [ ]
- no scroll in docs when on mobile when coming from the main menu (unless reloading the page) (related to SideMenu) [x]
- travis fails because of [browser global (location)](https://www.gatsbyjs.org/docs/debugging-html-builds/#how-to-check-if-code-classlanguage-textwindowcode-is-defined) [x]

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -1,198 +1,96 @@
...
+
+      const slugArray = slug.split('/');
+
+      nav.chapters.forEach(chapter => {
+        if (slugArray[2] === chapter.path) {
+          chapter.items.forEach((item, index) => {
+            if (index === 0) {
+              next.slug = `/docs/${slugArray[2]}/${chapter.items[index + 1].id}/`;
+              next.title = chapter.items[index + 1].title;
+            } else if (slugArray[3] === item.id) {
```
Comment (alanpoulain) [inline gatsby-node.js]: Return early instead of using an `else if`?

File: gatsby-node.js [hunk 2]
```diff
@@ -1,198 +1,96 @@
...
+              previous.slug =
+                chapter.items[index - 1].id === 'index'
+                  ? `/docs/${slugArray[2]}/`
+                  : `/docs/${slugArray[2]}/${chapter.items[index - 1].id}/`;
+              previous.title = chapter.items[index - 1].title;
+              if (chapter.items.length - 1 !== index) {
+                next.slug = `/docs/${slugArray[2]}/${chapter.items[index + 1].id}/`;
+                next.title = chapter.items[index + 1].title;
+              } else {
```
Comment (alanpoulain) [inline gatsby-node.js]: Return early instead of using `else`?

File: src/html.js [hunk 3]
```diff
@@ -1,37 +1,52 @@
...
           <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
           {this.props.headComponents}
-          {css}
           <link rel="stylesheet" href="https://cdn.jsdelivr.net/docsearch.js/2.5/docsearch.min.css" />
         </head>
-        <body>
+        <body {...this.props.bodyAttributes}>
           {this.props.preBodyComponents}
-          <div key={'body'} id="___gatsby" dangerouslySetInnerHTML={{ __html: this.props.body }} />
+          <div key={`body`} id="___gatsby" dangerouslySetInnerHTML={{ __html: this.props.body }} />
```
Comment (alanpoulain) [inline src/html.js]:
```suggestion
          <div key="body" id="___gatsby" dangerouslySetInnerHTML={{ __html: this.props.body }} />
```

**Comment by dunglas** [comment]
Thank you very much @Fabious!

**Pull request merged**